docker/engine.Dockerfile

```
@@ -56,6 +56,16 @@ COPY --from=sources /opt/engine-src   /opt/engine-src
 COPY --from=sources /tmp/smg-src      /opt/smg-src
 COPY --from=sources /tmp/scripts/     /tmp/scripts/
 
+# NGC TRT-LLM bases (1.3.0rc18) ship a debian-owned PyYAML with no pip RECORD
+# file, which pip refuses to uninstall when install-smg.sh resolves smg's
+# pyyaml dependency ("uninstall-no-record-file"). Shadow it with a pip-managed
+# copy (--ignore-installed leaves the distro files alone); the smg install can
+# then upgrade pyyaml normally. Scoped to trtllm so the other engine bases
+# keep their exact pip behavior.
+RUN if [ "${ENGINE}" = "trtllm" ]; then \
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)